client/components/tag/index.js

```diff
@@ -3,63 +3,104 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
 import classnames from 'classnames';
-import Gridicon from 'gridicons';
-import { IconButton } from '@wordpress/components';
+import { Button, Dashicon, IconButton, Popover } from '@wordpress/components';
 import PropTypes from 'prop-types';
-import { withInstanceId } from '@wordpress/compose';
+import { withState, withInstanceId } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 
 /**
- * This component can be used to show an item styled as a "tag", optionally with an `X` + "remove".
- * Generally this is used in a collection of selected items, see the Search component.
+ * This component can be used to show an item styled as a "tag", optionally with an `X` + "remove"
+ * or with a popover that is shown on click.
  *
  * @return { object } -
  */
-const Tag = ( { id, instanceId, label, remove, removeLabel, screenReaderLabel, className } ) => {
-	screenReaderLabel = screenReaderLabel || label;
-	if ( ! label ) {
-		// A null label probably means something went wrong
-		// @todo Maybe this should be a loading indicator?
-		return null;
-	}
-	const classes = classnames( 'woocommerce-tag', className, {
-		'has-remove': !! remove,
-	} );
-	const labelId = `woocommerce-tag__label-${ instanceId }`;
-	return (
-		<span className={ classes }>
-			<span className="woocommerce-tag__text" id={ labelId }>
+const Tag = withState( {
```

Comment:
Adding `withState` up here is breaking the docs generation script – instead of wrapping an anon function here, try wrapping `withInstanceId( Tag )` at the export.

Reply:
Interesting, thanks for pointing that out Kelly :til: